server/src/main/java/org/elasticsearch/script/StringFieldScript.java

```diff
@@ -28,8 +29,8 @@ public abstract class StringFieldScript extends AbstractFieldScript {
 
     public static final StringFieldScript.Factory PARSE_FROM_SOURCE = new Factory() {
         @Override
-        public LeafFactory newFactory(String field, Map<String, Object> params, SearchLookup lookup) {
-            return ctx -> new StringFieldScript(field, params, lookup, ctx) {
+        public LeafFactory newFactory(String field, Map<String, Object> params, SearchLookup lookup, OnScriptError onScriptError) {
+            return ctx -> new StringFieldScript(field, params, lookup, OnScriptError.FAIL, ctx) {
```

Comment:
> The controversial aspect here is that we are introducing support for the flag for scriptless runtime fields too, yet it has no effect on them.

Scratch that. You can only specify `on_script_error` when a `script` is also provided, so we are good. It may be confusing for users that script-less runtime fields can't be configured to throw error, but it kind of makes sense as effectively there is no script provided for them. We may want to clarify the docs around this.

Reply:
I raised https://github.com/elastic/elasticsearch/pull/92550 to clean up some of the existing ad-hoc try catch in favour of reusing the newly added centralized error handling.

Reply:
Speaking of documentation, I think it may make sense to explain in the docs what happens when a script fails after some values have been successfully emitted .